ginivision/src/main/java/net/gini/android/vision/internal/fileimport/FileChooserActivity.java

```diff
@@ -241,33 +282,50 @@ private void populateFileProviders() {
                 new ProvidersAppItemSelectedListener() {
                     @Override
                     public void onItemSelected(@NonNull final ProvidersAppItem item) {
-                        LOG.info("Requesting read storage permission");
-                        requestStoragePermission(new PermissionRequestListener() {
-                            @Override
-                            public void permissionGranted() {
-                                LOG.info("Read storage permission granted");
-                                launchApp(item);
-                            }
-
-                            @Override
-                            public void permissionDenied() {
-                                LOG.info("Read storage permission denied");
-                                showStoragePermissionDeniedDialog();
-                            }
-
-                            @Override
-                            public void shouldShowRequestPermissionRationale(
-                                    @NonNull final RationaleResponse response) {
-                                LOG.info("Show read storage permission rationale");
-                                showStoragePermissionRationale(response);
-                            }
-                        });
-
-
+                        requestStoragePermissionForAppItem(item, true);
                     }
                 }));
     }
 
+    private void requestStoragePermissionForAppItem(final ProvidersAppItem item,
+            final boolean allowRationale) {
```

Comment:
I know 😄 I've gave in to the temptation. Will change it.

Reply:
On second thought do you think it's so bad to keep the flag? Getting rid of it will require more methods to avoid duplication and I am not 100% sure it's worth the additional code in this case...

Reply:
It's not bad, it's just not that good. I always think flags make the method do more than one things which might lead to side effects and makes also unit testing a nightmare.